File: front/src/utils/localAssets.ts
const LOCAL_ASSET_PREFIXES = ['/assets/', '/uploads/']

/**
 * 封装 localAssetUrl 对应的前端业务处理步骤，供当前模块统一调用。
 * 参数：`value` 表示该步骤所需的业务参数。
 * 返回：返回该步骤计算、查询或校验后的结果。
 * 副作用：不直接修改持久化数据。
 */
export function localAssetUrl(value: unknown): string {
  const path = String(value || '').trim()
  if (!path || path.startsWith('//') || path.includes('\\') || path.split('/').includes('..')) return ''
  return LOCAL_ASSET_PREFIXES.some((prefix) => path.startsWith(prefix)) ? path : ''
}

/**
 * 封装 requireLocalAssetUrl 对应的前端业务处理步骤，供当前模块统一调用。
 * 参数：`value` 表示该步骤所需的业务参数。
 * 返回：返回该步骤计算、查询或校验后的结果。
 * 副作用：不直接修改持久化数据。
 */
export function requireLocalAssetUrl(value: unknown): string {
  const path = String(value || '').trim()
  const normalized = localAssetUrl(path)
  if (path && !normalized) throw new Error('只能使用 /assets/ 或 /uploads/ 下的本地资源路径，禁止引用外网地址')
  return normalized
}

File: backend/ops/local_assets.py
from rest_framework import serializers


LOCAL_ASSET_PREFIXES = ('/assets/', '/uploads/')


def normalize_local_asset_path(value, *, raise_error=False):
    """校验并规范化本地静态资源路径，拒绝外网地址。

    参数：`value` 表示待处理的输入值；`raise_error` 表示该步骤所需的raise_error 参数。
    返回：返回该业务步骤生成、查询或校验后的结果。
    副作用：不直接修改持久化数据。
    """
    path = str(value or '').strip()
    if not path:
        return ''
    valid = (
        path.startswith(LOCAL_ASSET_PREFIXES)
        and not path.startswith('//')
        and '\\' not in path
        and '..' not in path.split('/')
    )
    if valid:
        return path
    if raise_error:
        raise serializers.ValidationError('只能使用 /assets/ 或 /uploads/ 下的本地资源路径，禁止引用外网地址')
    return ''

